homeassistant/components/plex/server.py

```diff
@@ -367,3 +374,157 @@ def create_playqueue(self, media, **kwargs):
     def fetch_item(self, item):
         """Fetch item from Plex server."""
         return self._plex_server.fetchItem(item)
+
+    def lookup_media(self, media_type, **kwargs):
+        """Lookup a piece of media."""
+        media_type = media_type.lower()
+
+        if media_type == DOMAIN:
+            key = kwargs["plex_key"]
+            try:
+                return self.fetch_item(key)
+            except plexapi.exceptions.NotFound:
+                _LOGGER.error("Media for key %s not found", key)
+                return None
+
+        if media_type == MEDIA_TYPE_PLAYLIST:
+            try:
+                playlist_name = kwargs["playlist_name"]
+                return self.playlist(playlist_name)
+            except KeyError:
+                _LOGGER.error("Must specify 'playlist_name' for this search")
+                return None
+            except NotFound:
+                _LOGGER.error(
+                    "Playlist '%s' not found", playlist_name,
+                )
+                return None
+
+        try:
+            library_name = kwargs["library_name"]
+            library_section = self.library.section(library_name)
+        except KeyError:
+            _LOGGER.error("Must specify 'library_name' for this search")
+            return None
+        except NotFound:
+            _LOGGER.error("Library '%s' not found", library_name)
+            return None
+
+        def lookup_music(**kwargs):
```

Comment:
The nesting here is a bit much.  Can these be broken out ?

Reply:
Good point. No need to pass `kwargs` as a parameter. Removed.